### PR TITLE
Sync parameter order in boston_housing with reuters

### PR DESCRIPTION
**Pull request by ozabluda**
Those are the only two that have `test_split`. Syncing in that direction, because `boston_housing` is not referenced anywhere in Keras code, while `reuters` is. I think there was some discussion of datasets API in tensorflow, but can't find it now. Should `test_split` be renamed `validation_split` for consistency with the rest of codebase?

### CODE REVIEW

**Comment by ozabluda** [comment]
FWTW, `seed` parameter goes after the analog of `split` (matching this PR) in 
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/learn/python/learn/datasets/mnist.py
